api/src/test/java/org/openmrs/module/sync2/client/rest/RESTClientHelperTest.java

```diff
@@ -53,23 +79,41 @@ public void deleteRequest() throws Exception {
 	public void updateRequest() throws Exception {
 		Patient patient = new Patient();
 		patient.setUuid(TEST_PATIENT_UUID);
-		RestResource patientResource = RestResourceCreationUtil.createRestResourceFromOpenMRSData(patient);
-		RequestEntity expected = new RequestEntity(patientResource, HttpMethod.POST,
+		SimpleObject simpleObject = getSimpleObject(patient);
+		RequestEntity expected = new RequestEntity(simpleObject, HttpMethod.POST,
 				URI.create(TEST_URI + "/" + TEST_PATIENT_UUID));
 
-		RESTClientHelper restClientHelper = new RESTClientHelper();
-		assertEquals(expected, restClientHelper.updateRequest(TEST_URI, patient));
+		assertEquals(expected, restClientHelper.updateRequest(TEST_URI, simpleObject));
 	}
 
 	@Test
 	public void resolveCategoryByCategory() {
 		RESTClientHelper restClientHelper = new RESTClientHelper();
-		assertEquals(org.openmrs.module.sync2.client.rest.resource.Patient.class,
-				restClientHelper.resolveCategoryByCategory(CATEGORY_PATIENT));
-		assertEquals(org.openmrs.module.sync2.client.rest.resource.Visit.class,
-				restClientHelper.resolveCategoryByCategory(CATEGORY_VISIT));
-		assertEquals(org.openmrs.module.sync2.client.rest.resource.Encounter.class,
-				restClientHelper.resolveCategoryByCategory(CATEGORY_ENCOUNTER));
+		assertEquals(SimpleObject.class, restClientHelper.resolveCategoryByCategory(CATEGORY_PATIENT));
+		assertEquals(SimpleObject.class, restClientHelper.resolveCategoryByCategory(CATEGORY_VISIT));
+		assertEquals(AuditMessage.class, restClientHelper.resolveCategoryByCategory(CATEGORY_AUDIT_MESSAGE));
+	}
+
+	private SimpleObject getSimpleObject(Object object) throws IOException {
+		String json = getJson(object);
+		return SimpleObject.parseJson(json);
+	}
+
+	private String getJson(Object object) {
+		Gson defaultJsonParser = createDefaultGson();
+		return defaultJsonParser.toJson(object);
+	}
+
+	private Gson createDefaultGson() {
```

Comment:
Please move createDefaultGson to constants and use the same Gson in the business logic and in tests

Reply:
Done